tools/module_diag_hailcast.F90

```diff
@@ -791,6 +791,7 @@ SUBROUTINE hailstone_driver ( ii,jj,TCA, h1d, ht, PA, rho1d,&
             KBAS = k
          ENDIF
     ENDDO
+    KBAS = MAX(KBAS, 2)     ! start at least from 2 to avoid cloud base at k=1, since then k-1 would be zero and out of bounds
```

Comment:
The enclosed loop not only computes `KBAS` and it also fills the one dimension arrays `RIA` & `RWA`, which are used later in the code (including `RIA(1)` & `RWA(1)`). That is why starting the loop from `k=2` does not seem a good solution because it will leave `RIA(1)` & `RWA(1)` uninitialized.

Reply:
I see now. Then your solution is the best option. Thanks.